### DISCUSSION TITLE
MJX: Advice on what might be causing mesh humanoid foot slipping

### DISCUSSION THREAD

**Discussion by areiner222** [google-deepmind/mujoco#1942] [Asking for Help]
Hi,

I'm trying to use MuJoCo - MJX for humanoid imitation learning, and looking for some advice on some foot sliding/hopping I'm experiencing.
I've noticed that this occurs with some orientations of the foot and hoping to get a gut reaction on if I could tune anything specifically to improve this? 

I'm building a mujoco model based on the SMPL-X mesh whereby I do body part segmentation followed by using [coacd](https://github.com/SarahWeiii/CoACD) to further decompose the body part meshes to be approximately convex.  

I've attached a video of the behavior:
https://github.com/user-attachments/assets/0fed3ec9-867e-4461-9912-36f270f95157

I've experimented with a wide range of impratios (not totally sure if this is the right way to go) and the behavior persists.

Appreciate any advice including if I should be reconsidering how I represent the foot mesh! (have considered using primitives instead but since I build the mesh on the fly with the smpl shape parameters I think I'd have to functionally map the mesh to the scale of the primitive collection - probably doable but have not thought through).

**Comment by btaba** [comment]
Hi @areiner222 , here are a couple of suggestions:

1. Visualize the contacts by typing `c` in simulate. This should give you an indication if the foot mesh is generating stable contact points or not.
2. The more likely issue is that you need to keep tuning the contact solver parameters. It looks like the contacts are perhaps a bit too soft (resulting in bouncing). Consider reducing the timeconst on [`solref`](https://mujoco.readthedocs.io/en/stable/modeling.html#reference), you'll also need to lower `opt.timestep`, and may want to tune `opt.iterations` and `opt.ls_iterations` while you're at it to make sure the sim is stable (we usually set solver iters really low in MJX, but that may or may not work for your use-case).
3. For sliding, consider fiddling with `friction`, elliptic `opt.cone`, `condim`, and `impratio` as you were doing.

Let us know if there are any updates!
